src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,162 @@
+        }
+
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = null;
+        String sql = "SELECT * FROM books WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            if (resultSet.next()) {
+                book = new Book();
+                Long rowId = resultSet.getObject("id", Long.class);
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                book.setId(rowId);
+                book.setTitle(title);
+                book.setPrice(price);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant create connection", e);
+        }
+
+        Optional<Book> result = Optional.ofNullable(book);
+        return result;
+    }
+
+    @Override
+    public Book update(Book book) {
+        if (book == null || book.getId() == null) {
+            throw new RuntimeException("unacceptable data format");
+        }
+        Book currentBook = findById(book.getId()).orElseThrow(() -> new RuntimeException(
+                "There is no object with such id: " + book.getId()));
+
+        if (book.equals(currentBook)) {
+            return book;
+        }
+
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+            preparedStatement.executeUpdate();
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant create connection", e);
+        }
+
+        currentBook = findById(book.getId()).get();
+        if (currentBook.equals(book)) {
+            return currentBook;
+        }
+        throw new RuntimeException("Update failed");
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        if (findById(id).isEmpty()) {
+            return false;
+        }
+
+        String sql = "DELETE FROM books WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            preparedStatement.executeUpdate();
+
+            if (findById(id).isPresent()) {
+                return false;
+            }
+
+            return true;
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant create connection", e);
```

Comment:
The error message 'Cant create connection' is misleading here. It should reflect the operation being performed, such as 'Failed to delete book'.